samcli/local/apigw/local_apigw_service.py

```diff
@@ -405,7 +409,14 @@ def _parse_v1_payload_format_lambda_output(lambda_output: str, binary_types, fla
         body = json_output.get("body")
         if body is None:
             LOG.warning("Lambda returned empty body!")
-        is_base_64_encoded = json_output.get("isBase64Encoded") or False
+
+        # Http API gateway checks lambda response for isBase64Encoded field, and ignore base64Encoded
+        # Rest API gateway checks first the field base64Encoded field, if not exist, it checks isBase64Encoded field
```

Comment:
Is this true? Docs suggests isBase64Encoded is only supported: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format

Reply:
based on my trials that I found the following (as I could not find in the documentation anything related if APi gateway will fail if the lambda response contains any unknown field or not)
- for HTTP API Gateway just skip any un known fields 
- for Rest API, it check also for base64Encoded and fail if there is any other unknown fields .. and for doing the decoding it depends on the value of base64Encoded first, then isBase64Encoded ... which means if both fields are there and base64Encoded = True, and isBase64Encoded = False .. Api gateway will try to decode the response body

Reply:
done